src/domain-engine.js

```diff
@@ -85,17 +85,30 @@ class DomainEngine {
     return domain
   }
 
-  dispatch(state: Object, action: Action): Object {
+  dispatch(action: Action, state: Object, snapshot: Object) {
```

Comment:
This is cool Nate. I'm assuming the `dispatch` method is what actually sends actions (and their payloads) to the store. What exactly does the handler do here?